### DISCUSSION TITLE
Armv7E-M Arch

### DISCUSSION THREAD

**Discussion by vdemcak** [daniel5151/gdbstub#139] [Q&A]
Hello, 
I'm currently in the process of creating a custom gdbstub target for STM32WL that has a ARM Cortex-M4 running on Armv7E-M architecture. I was unable to find the `Arch` for it in this crate; are there any plans on implementing it or does a similiar crate containing it already exist?

Thanks.

**Comment by daniel5151** [comment]
As per the [docs page](https://docs.rs/gdbstub_arch/latest/gdbstub_arch/) of `gdbstub_arch`: the crate is a collection of _community_ created/tested `Arch` impls, so if you don't see an impl for your particular architecture, you're more than welcome to write one yourself + open a PR to upstream it.

Unless someone else happens to be interested in the exact same `Arch` as you are, I suspect you'll need to roll up your sleeves and implement it yourself :)
